src/Common/ProfileEvents.h

```diff
@@ -60,7 +60,7 @@ namespace ProfileEvents
         Counter * counters = nullptr;
         std::unique_ptr<Counter[]> counters_holder;
         /// Used to propagate increments
-        Counters * parent = nullptr;
```

Comment:
This `parent` field is assigned in [ProcessList::insert()](https://github.com/ClickHouse/ClickHouse/blob/e3e6905638251e93a249322801002bdc9c132b22/src/Interpreters/ProcessList.cpp#L251-L252) similar to how it is done for `MemoryTracker`, and the `parent` field in `MemoryTracker` is atomic, so I think we can do the `parent` field atomic for `Counters` too.